corehq/form_processor/abstract_models.py

```diff
@@ -215,6 +215,14 @@ def to_api_json(self):
     def set_case_id(self, case_id):
         raise NotImplementedError()
 
+    @property
+    def server_opened_on(self):
+        try:
+            open_action = self.actions[0]
```

Comment:
Right. Ok I guess in that case I'll just calculate the `server_opened_on` for my particular report, and not make it available on the model in general. In my case, the cases should almost always have just a single case action.